spacy/ml/models/entity_linker.py

```diff
@@ -78,23 +80,36 @@ def span_maker_forward(model, docs: List[Doc], is_train) -> Tuple[Ragged, Callab
 
 
 @registry.misc("spacy.KBFromFile.v1")
-def load_kb(kb_path: Path) -> Callable[[Vocab], KnowledgeBase]:
-    def kb_from_file(vocab):
-        kb = KnowledgeBase(vocab, entity_vector_length=1)
+def load_kb(
+    kb_path: Path,
+) -> Callable[[Vocab], InMemoryLookupKB]:
```

Comment:
I think we shouldn't change the type of this existing registry function, to avoid any kind of issue with type checking in user's applications. We could in time retire this function for a new version with a naming that makes it more clear that it's the `InMemoryLookupKB` that is returned.